spec/system/consumer/registration_spec.rb

```diff
@@ -157,6 +157,54 @@
       expect(page).to have_content "Your profile live"
     end
 
+    context "Enterprise name is already taken" do
+      let(:owner) do
+        Spree::User.create!(email: "penny.profile@example.org", password: "cannotbeblank")
+      end
+
+      before do
+        address = Spree::Address.create!(firstname: 'John', lastname: 'Doe',
+                                         address1: '1400 Sesame street', zipcode: '3070',
+                                         city: 'Southcote', phone: '12 3456 7890',
+                                         country_id: 1, state_id: 1, company: 'unused')
+        Enterprise.create(name: 'My Awesome Enterprise', address:, owner:)
+      end
+
+      it "checks that button after failure is still enabled" do
+        visit registration_path
+        switch_to_login_tab
+
+        fill_in "Email", with: user.email
+        fill_in "Password", with: user.password
+        click_button "Login"
+        click_button "Let's get started!"
+
+        fill_in 'enterprise_name', with: "My Awesome Enterprise"
+        fill_in 'enterprise_address', with: '123 Abc Street'
+        fill_in 'enterprise_city', with: 'Northcote'
+        fill_in 'enterprise_zipcode', with: '3070'
+        select 'Vic', from: 'enterprise_state'
+        click_button "Continue"
+
+        fill_in 'enterprise_contact', with: 'Saskia Munroe'
+        fill_in 'enterprise_phone', with: '12 3456 7890'
+
+        click_button "Continue"
+        click_button "Create Profile"
+        click_link "producer-panel"
+        alert_text = <<~TEXT.strip
+          Name has already been taken. If this is your enterprise and you would \
+          like to claim ownership, or if you would like to trade with this \
+          enterprise please contact the current manager of this profile at \
+          #{owner.email}.
```

Comment:
You defined the email above and therefore it's much clearer to use that in plain text.
```suggestion
          penny.profile@example.org.
```